query/query.go

```diff
@@ -70,6 +70,7 @@ type Query struct {
 // Entry is a query result entry.
 type Entry struct {
 	Key        string    // cant be ds.Key because circular imports ...!!!
+	Size       int       // Might be zero if the datastore doesn't support listing the size with KeysOnly
```

Comment:
So, a value can _also_ just be empty. We'd have to use -1 to mean "not set" (or a `HasSize` flag).

What if we:

1. Added a field to the query to _request_ the size (like we do with expirations).
2. Always included the size if requested, potentially doing extra work.

This is a breaking change for datastore implementations but it's a pretty minor breaking change.

Reply:
That works for me. A datastore layer can also hijack the query and enable the flag if it's needed or considered free.